google-account-plugin/src/com/google/gct/login/GoogleLogin.java

```diff
@@ -567,6 +567,16 @@ public String getInfo() {
   private class AndroidUiFacade implements UiFacade {
     private GoogleLoginActionButton myButton;
     private volatile CancellableServerReceiver receiver = null;
+    private GoogleLoginMessageExtender messageExtender;
+
+    public AndroidUiFacade() {
+      GoogleLoginMessageExtender[] messageExtenders = Extensions.getExtensions(GoogleLoginMessageExtender.EP_NAME);
+      if (messageExtenders.length == 0) {
+        messageExtender = new DummyGoogleLoginMessageExtender();
+      } else {
+        messageExtender = messageExtenders[0];
```

Comment:
I don't think that's a safe assumption.  If you include an EP in the
account plugin many of the plugin's clients in Android Studio (such as
Google Play, Cloud Testing, Firebase), could conceivably try to use it.


On Tue, Feb 2, 2016 at 11:48 AM, akerekes <notifications@github.com> wrote:

> In google-account-plugin/src/com/google/gct/login/GoogleLogin.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/395#discussion_r51596031>
> :
>
> > @@ -567,6 +567,16 @@ public String getInfo() {
> >    private class AndroidUiFacade implements UiFacade {
> >      private GoogleLoginActionButton myButton;
> >      private volatile CancellableServerReceiver receiver = null;
> > +    private GoogleLoginMessageExtender messageExtender;
> > +
> > +    public AndroidUiFacade() {
> > +      GoogleLoginMessageExtender[] messageExtenders = Extensions.getExtensions(GoogleLoginMessageExtender.EP_NAME);
> > +      if (messageExtenders.length == 0) {
> > +        messageExtender = new DummyGoogleLoginMessageExtender();
> > +      } else {
> > +        messageExtender = messageExtenders[0];
>
> At that time it seemed reasonable to expect only one extension to provide
> messages. We can however allow any number, though their order of execution
> will be arbitrary initially (the order IJ/AS returns the corresponding
> extensions) as I see no point in implementing some kind of priority logic
> currently.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/395/files#r51596031>
> .
>